File: Intermediate/problem_9.py
'''
author : Pythonicboat
topic : program in python to bubble sort
'''

def bubbleSort(array):
  for i in range(len(array)):
    for j in range(0, len(array) - i - 1):

        #comparing adjacent elements
        if array[j] > array[j + 1]:

            # swapping elements
            temp = array[j]
            array[j] = array[j+1]
            array[j+1] = temp

data = [5, -34, 12, 7, -1]

bubbleSort(data)

print('Sorted Array in Ascending Order: ')
print(data)

File: Intermediate/C#/problem_9.c
/**
author : Pythonicboat
topic : program in c to bubble sort
**/

#include <stdio.h>

void bubbleSort(int array[], int size) {
  for (int step = 0; step < size - 1; ++step) {
    for (int i = 0; i < size - step - 1; ++i) {

      // comparing adjacent elements
      if (array[i] > array[i + 1]) {

        // swapping elements
        int temp = array[i];
        array[i] = array[i + 1];
        array[i + 1] = temp;
      }
    }
  }
}

void printArray(int array[], int size) {
  for (int i = 0; i < size; ++i) {
    printf("%d  ", array[i]);
  }
  printf("\n");
}

int main() {
  int data[] = {5, -34, 12, 7, -1};
  int size = sizeof(data) / sizeof(data[0]);

  bubbleSort(data, size);

  printf("Sorted Array in Ascending Order : ");
  printArray(data, size);
}
